internal/tjson/schema.go

```diff
@@ -240,11 +241,34 @@ func DocumentSchema(doc *types.Document) (*Schema, error) {
 	return &schema, nil
 }
 
+// subdocumentSchema returns a JSON Schema for the given subdocument.
+// Subdocument is a "nested" document that can be used as a property of another document or subdocument.
```

Comment:
Let me know if you have a better idea for this case.

Reply:
LGTM